src/cli/cli.cpp

```diff
@@ -1121,12 +1121,13 @@ void Interpreter::ProcessEidCache(int argc, char *argv[])
 {
     OT_UNUSED_VARIABLE(argc);
     OT_UNUSED_VARIABLE(argv);
+    otError error = OT_ERROR_NONE;
 
     otEidCacheEntry entry;
 
     for (uint8_t i = 0;; i++)
     {
-        SuccessOrExit(otThreadGetEidCacheEntry(mInstance, i, &entry));
+        SuccessOrExit(error = otThreadGetEidCacheEntry(mInstance, i, &entry));
```

Comment:
I think not checking error here is intentional, otherwise will always get to `AppendResult()` with an error
The `for` loop keeps going till we get an error from `otThreadGetEidCacheEntry()` which indicates the index `i` is out of bound and we exit the loop.
https://github.com/openthread/openthread/blob/a75b412ecfe3e3c286c57517b08171eb05fef0c4/include/openthread/thread_ftd.h#L509-L520

Just as a note, I am changing the public API for address cache table from this open PR https://github.com/openthread/openthread/pull/4599 which will use an iterator and return `OT_ERROR_NOT_FOUND` when we reach last cache entry in the list.